pinot-integration-tests/src/test/java/org/apache/pinot/compat/tests/AvroRowExtractor.java

```diff
@@ -0,0 +1,145 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.compat.tests;
+
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nullable;
+import org.apache.avro.Schema;
+import org.apache.avro.generic.GenericData;
+import org.apache.avro.generic.GenericRecord;
+import org.apache.pinot.spi.data.readers.GenericRow;
+
+
+/**
+ * Extract avro record from generic row
+ */
+public class AvroRowExtractor {
```

Comment:
Can we use AvroRecordExtractor? We can add the avro package as dependency here?

Reply:
The `AvroRecordExtractor` extracts data from `GenericRecord` to `GenericRow`. In this case, we actually need to extracts data from `GenericRow` to `GenericRecord`, so it cannot be resused.